### PR TITLE
Finished tests for transform : Caleb

### LINKED ISSUE #14: Create test file for weekly astronomy transform script
As a developer, I want to test the transform.py file to ensure that the data is being processed correctly and robustly, and that the functions deal with appropriate edge cases and erroneous input effectively so that the user experience is as intended.

### LINKED ISSUE #196: Write tests for weekly astronomy transform
As a developer, I want to test the transform.py file to ensure that the data is being processed correctly and robustly, and that the functions deal with appropriate edge cases and erroneous input effectively so that the user experience is as intended.

- Make sure coverage is at least 75%
- Use pytest

### PR DESCRIPTION
**Pull request by CalebMathers**
This closes #14 
This closes #196 

Finished tests or astronomy transform script. All astronomy scripts tested now; total coverage 95%

Also added a few pylint disables to fix issues.

### CODE REVIEW

File: astronomy-pipeline/test_astronomy_transform.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
+'''Tests for the astronomy transform script.'''
```
Comment (MurasatoNaoya) [inline astronomy-pipeline/test_astronomy_transform.py:1]: This is an extensive test suit for testing the wide range of functionality in the astronomy_transform module. Good job.

File: astronomy-pipeline/conftest.py [hunk 2]
```diff
@@ -140,3 +142,17 @@ def astronomy_data_dict():
         'positions'], 'moon_phase_list': ['moon']}
 
     return astronomy_data
+
+
+@pytest.fixture
+def position_dataframe_example():
```
Comment (AlexOBarnes) [inline astronomy-pipeline/conftest.py:148]: Good use of pytest fixtures

File: astronomy-pipeline/test_astronomy_transform.py [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+import pytest
+
+from astronomy_transform import (transform_astronomy_data, load_from_file,
+                                 get_data_into_dataframe, get_body_mapping,
+                                 get_constellation_mapping, clean_position_data,
+                                 get_moon_df, convert_positions_datetime,
+                                 convert_moon_datetime)
+
+
+class TestLoadFromFile():
```
[the remark this answers is not in the document]
Comment (AlexOBarnes) [reply astronomy-pipeline/test_astronomy_transform.py:16]: Very nice for readability

File: astronomy-pipeline/test_astronomy_transform.py [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+    '''Tests for the load from file function.'''
+
+    @patch('json.load')
+    @patch('builtins.open')
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_transform.py:20]: Good for idempotency

File: astronomy-pipeline/test_astronomy_transform.py [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+        mock_cursor = MagicMock()
+        mock_get_db_connection.return_value.__enter__.return_value = mock_conn
+        mock_conn.cursor.return_value = mock_cursor
+
+        result = get_constellation_mapping()
+
+        assert isinstance(result, dict)
+
+
+class TestCleanPositionData():
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_transform.py:162]: Very comprehensive test suite

File: astronomy-pipeline/test_astronomy_transform.py [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+        mock_clean_position_data.return_value = MagicMock()
+        mock_convert_moon_datetime.return_value = MagicMock()
+        mock_get_moon_df.return_value = MagicMock()
+        mock_convert_positions_datetime.return_value = MagicMock()
+        mock_get_data_into_dataframe.return_value = MagicMock()
+
+        result = transform_astronomy_data({'moon_phase_urls': 'test'})
+
+        assert 'positions_list' in result
+        assert 'moon_phase_list' in result
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_transform.py:318]: Do we have a test coverage?

**Review by AlexOBarnes — APPROVED** [review]
Looks good to me

File: astronomy-pipeline/conftest.py [hunk 7]
```diff
@@ -140,3 +142,17 @@ def astronomy_data_dict():
         'positions'], 'moon_phase_list': ['moon']}
 
     return astronomy_data
+
+
+@pytest.fixture
```
Comment (MurasatoNaoya) [inline astronomy-pipeline/conftest.py:147]: A good, extensive list of fixtures for later patching.

File: astronomy-pipeline/test_astronomy_transform.py [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+        mock_cursor = MagicMock()
+        mock_get_db_connection.return_value.__enter__.return_value = mock_conn
+        mock_conn.cursor.return_value = mock_cursor
+
+        get_body_mapping()
+
+        mock_cursor.execute.assert_called_once_with(
+            'SELECT body_id, body_name from body;')
+
+    @patch('astronomy_transform.get_db_connection')
```
Comment (MurasatoNaoya) [inline astronomy-pipeline/test_astronomy_transform.py:105]: Good mocking of database connections to avoid interacting with our RDS for testing.

**Review by MurasatoNaoya — APPROVED** [review]
LGTM.
